packages/melody-compiler/src/convert/blocks.js

```diff
@@ -246,6 +246,15 @@ export default {
     },
     IncludeStatement: {
         exit(path) {
+            const sourceType = path.node.source.type;
+            if (sourceType !== 'StringLiteral') {
+                this.error(
```

Comment:
The error method can be used to create really nice error messages. Take a look at the method itself here to gain a better understanding of the parameters: https://github.com/trivago/melody/blob/master/packages/melody-compiler/src/state/State.js#L69

You can take a look at another usage here:
https://github.com/trivago/melody/blob/437ea7e0845f42d1bac2dc38e9d39967204e3c4b/packages/melody-compiler/src/finalizer/validate.js#L50

to get a feeling for the quality of error message we want to present to our users. This is basically our only chance to reduce the frustration of our engineers when they encounter this issue by providing them with guidance and helping them to find other solutions to their problem.

For the title, I'd suggest something along the lines of "Dynamic includes are not supported". The body could include a hint of how to avoid the problem.

Reply:
ok, i'll work on it

Reply:
The new async-mount feature supports dynamic includes. Do you think it'd make sense to mention that in this error as well?

Reply:
Yes it makes perfectly sense. I'll add it.